scripts/DESeq_functions.R

```diff
@@ -0,0 +1,235 @@
+        
+        # sanity checks
+        stopifnot(exprs = {
+            nrow(DESeqDesign_subset) > 0
+        })
+
+        Filter <- matrix(data = NA, ncol = 3, nrow = nrow(Counts))
+        rownames(Filter) <- rownames(Counts) # genes
+        colnames(Filter) <- c("Low", "quantile", "spike")
+
+        # Apply the "Relevance" condition
+        message(paste0("Filtering genes: 75% of at least 1 group need to be above ", params$MinCount, " CPM"))
+        SampPerGroup <- table(DESeqDesign_subset[, design])
+
+        for (gene in 1:nrow(dds)) {
+            CountsPass <- NULL
+            for (group in 1:length(SampPerGroup)) { 
+                sampleCols <- grep(dimnames(SampPerGroup)[[1]][group], DESeqDesign_subset[, design])
+                Check <- sum(CPMdds[gene,sampleCols] >= params$MinCount) >= 0.75 * SampPerGroup[group]
+                CountsPass <- c(CountsPass, Check)
+            }
+            if (sum(CountsPass) > 0) {Filter[gene, 1] <- 1 }  else {Filter[gene,1] <- 0 }
+        }
+
+        compte <- Counts[Filter[,1] == 1,]
+        Filter <- Filter[rownames(Filter) %in% rownames(compte), , drop = F]
+
+        message(paste0("Relevance filtering removed ", nrow(dds) - nrow(Filter),
+                    " genes from the ", nrow(dds)," assessed. ",
+                    nrow(Filter), " genes remaining"))
+
+        message("Obtaining the DESeq2 results")
+
+        currentContrast <- c(design, condition2, condition1)
+        bpparam <- MulticoreParam(params$cpus)
+
+        res <- DESeq2::results(dds[rownames(compte),],
+                                parallel = TRUE,
+                                BPPARAM = bpparam,
+                                contrast = currentContrast,
+                                pAdjustMethod = 'fdr',
+                                cooksCutoff = params$cooks) # If Cooks cutoff disabled - manually inspect.
+
+        res <- lfcShrink(dds = dds[rownames(compte),],
+                        contrast = currentContrast,
+                        res = res,
+                        type = "ashr")
+
+        resListAll[[contrast_string]] <- res
+
+
+        # Create output tables
+        norm_data <- counts(dds[rownames(compte)], normalized = TRUE)
+        DEsamples <- subset(res, res$padj < params$alpha)
+        if (nrow(DEsamples) == 0) {
+            message("No significant results were found for this contrast. Moving on...")
+            next
+        }
+        DECounts <- compte[rownames(compte) %in% rownames(DEsamples), , drop = F]
+        Filter <- Filter[rownames(Filter) %in% rownames(DECounts), , drop = F]
+
+        message("Check median against third quantile" )
+        message("AND")
+        message("Check the presence of a spike" )
+
+        for (gene in 1:nrow(DECounts)) {
+            # Check the median against third quantile
+            quantilePass <- NULL
+            sampleColsg1 <- grep(dimnames(SampPerGroup)[[1]][1],DESeqDesign_subset[,design])
```

Comment:
These lines will need to be fixed to catch that ordering issue... can just copy and paste the chunk from the Rmd file from the bisphenols branch (I will merge that today hopefully)